### ISSUE TITLE
Bugs with `docs` spreads

### ISSUE THREAD

**Issue by bmish** [not-an-aardvark/eslint-plugin-eslint-plugin#633]
Extracting this from other issue:

> I have similar issues with `eslint-plugin/require-meta-docs-*` rules. Basicually if I move the `docs` definition out of the `meta` JSON, I cannot get any of the rules to pass e.g. both would fail:
> ```js
> const docs = require("./rule.docs.js");
> meta: {
>     docs
> } 
> ```
> and
> ```js
> const docs = require("./rule.docs.js");
> meta: {
>    docs: { ...docs }
> }
> ...
> 
> At the moment I work this around by manually adding every single property to the `docs` array. 

 _Originally posted by @tonyganchev in [#274](https://github.com/eslint-community/eslint-plugin-eslint-plugin/issues/274#issuecomment-3448134974)_
